wallet/public/index.html

```diff
@@ -60,19 +37,66 @@
       Learn how to configure a non-root public URL by running `npm run build`.
     -->
     <title>Agoric Wallet</title>
+    <script type="module">
+      const MAX_ATTEMPTS = 10;
+      const WAIT_PER_ATTEMPT = 10;
+      let attempts = 0;
+      const startWhenKeplrAvailable = () => {
+        console.debug('looking for keplr');
+        attempts += 1;
+        if (window.keplr) {
+          // Allow the React dev environment to extend the console for debugging
+          // features.
+          const consoleTaming =
+            '%NODE_ENV%' === 'production' ? 'safe' : 'unsafe';
+
+          const { pow: mathPow } = Math;
+          Math.pow = (base, exp) =>
+            typeof base === 'bigint' && typeof exp === 'bigint'
+              ? base ** exp
+              : mathPow(base, exp);
+
+          lockdown({
+            errorTaming: 'unsafe',
+            overrideTaming: 'severe',
+            consoleTaming: consoleTaming,
+          });
+          console.log('lockdown done.');
+
+          window.addEventListener('unhandledrejection', ev => {
+            ev.stopImmediatePropagation();
+          });
+
+          console.log('Importing app bundle');
+          import('/wallet/<%= compilation.getAssets()[0].name %>');
+        } else {
+          if (attempts < MAX_ATTEMPTS) {
+            setTimeout(startWhenKeplrAvailable, WAIT_PER_ATTEMPT);
+          } else {
+            const notice = document.getElementById('keplr-missing');
+            notice.hidden = false;
+          }
+        }
+      };
+      startWhenKeplrAvailable();
+    </script>
   </head>
+
   <body style="overflow-x: hidden; overflow-y: scroll">
     <noscript>You need to enable JavaScript to run this app.</noscript>
-    <div id="root"></div>
-    <!--
-      This HTML file is a template.
-      If you open it directly in the browser, you will see an empty page.
-
-      You can add webfonts, meta tags, or analytics to this file.
-      The build step will place the bundled scripts into the <body> tag.
-
-      To begin the development, run `npm start` or `yarn start`.
-      To create a production bundle, use `npm run build` or `yarn build`.
-    -->
+    <div id="root">
+      <!-- replaced by React mounting -->
+    </div>
+    <div id="keplr-missing" hidden>
+      <h1>Keplr required</h1>
```

Comment:
Nice enhancement

Reply:
I too am old and remember when progressive enhancement was cool. Love it.